holidays/countries/botswana.py

```diff
@@ -30,58 +29,61 @@ class Botswana(HolidayBase):
     country = "BW"
     special_holidays = {2019: ((JUL, 2, "Public Holiday"),)}
 
-    def _populate(self, year: int):
-        def _add_with_observed(
-            hol_date: date, hol_name1: str, hol_name2: str = None
-        ) -> None:
-            self[hol_date] = hol_name1
-            if hol_name2:
-                self[hol_date + td(days=+1)] = hol_name2
-
-            if self.observed and year >= 1995:
-                if self._is_saturday(hol_date) and hol_name2:
-                    self[hol_date + td(days=+2)] = f"{hol_name2} (Observed)"
-                elif self._is_sunday(hol_date):
-                    self[
-                        hol_date + td(days=+2 if hol_name2 else +1)
-                    ] = f"{hol_name1} (Observed)"
+    def __init__(self, *args, **kwargs) -> None:
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        super().__init__(*args, **kwargs)
+
+    def _add_observed(self, dt: date) -> None:
+        if self.observed and self._is_sunday(dt):
+            self._add_holiday(
+                "%s (Observed)" % self[dt],
+                dt + td(days=2 if dt + td(days=+1) in self else 1),
+            )
 
+    def _populate(self, year):
         if year <= 1965:
             return None
 
         super()._populate(year)
 
-        _add_with_observed(
-            date(year, JAN, 1), "New Year's Day", "New Year's Day Holiday"
+        jan_1 = self._add_new_years_day("New Year's Day")
```

Comment:
```
self._add_observed(
    self._add_new_years_day("New Year's Day")
)
```

for consistency?

Reply:
Here we need add observed for "New Year's Day" when "New Year's Day Holiday" already exist. When Jan 1 is Sunday, "New Year's Day (Observed)" should be Jan, 3 (Thuesday). And same for other 2-day holidays in Botswana.

Reply:
Would L82-85 syntax be acceptable here?

```
jan_1 = self._add_new_years_day("New Year's Day")
jan_2 = self._add_new_years_day_two("New Year's Day Holiday")
self._add_observed(jan_1)
self._add_observed(jan_2)
```

// sometimes observance is so tricky 🤯